default/python/AI/AIFleetMission.py

```diff
@@ -533,6 +545,11 @@ def issue_fleet_orders(self):
                     last_sys_target = self.target.id
                     debug("Check if PROTECT_REGION mission with target %d is finished.", last_sys_target)
 
+                # for SECURE missions, only release fleet if no settle or invade mission is ongoing
+                if self.type == MissionType.SECURE and self.target.id in secure_targets:
+                    debug(f"Secure mission for {self.target} not yet finished")
```

Comment:
Maybe print other info about the mission? For example fleet.   You can try to print `self`.  This will cal `__str__` method implicitly.

Reply:
There is lots of debug outputs about the mission before, reading the logs you do not miss that information.